server/monitor/subscriptions.go

```diff
@@ -0,0 +1,132 @@
+package monitor
+
+import (
+	"time"
+
+	"github.com/mattermost/mattermost-plugin-msteams-sync/server/msteams"
+	"github.com/mattermost/mattermost-plugin-msteams-sync/server/store/storemodels"
+)
+
+func (m *Monitor) checkChannelsSubscriptions() {
+	m.api.LogDebug("Checking for channels subscriptions")
+	subscriptions, err := m.store.ListChannelSubscriptionsToCheck()
+	if err != nil {
+		m.api.LogError("Unable to get the channel subscriptions", "error", err)
+		return
+	}
+	m.api.LogDebug("Refreshing channels subscriptions", "count", len(subscriptions))
+
+	for _, subscription := range subscriptions {
+		if time.Until(subscription.ExpiresOn) < (15 * time.Second) {
+			err := m.recreateChannelSubscription(subscription.SubscriptionID, subscription.TeamID, subscription.ChannelID, subscription.Secret)
+			if err != nil {
+				m.api.LogError("Unable to recreate channel subscription properly", "error", err)
+			}
+		} else {
+			err := m.refreshSubscription(subscription.SubscriptionID)
+			if err != nil {
+				m.api.LogError("Unable to refresh channel subscription properly", "error", err)
+			}
+		}
+	}
+}
+
+func (m *Monitor) checkChatsSubscriptions() {
+	m.api.LogDebug("Checking for chats subscriptions")
+	subscriptions, err := m.store.ListChatSubscriptionsToCheck()
+	if err != nil {
+		m.api.LogError("Unable to get the chat subscriptions", "error", err)
+		return
+	}
+	m.api.LogDebug("Refreshing chats subscriptions", "count", len(subscriptions))
+
+	for _, subscription := range subscriptions {
+		if time.Until(subscription.ExpiresOn) < (15 * time.Second) {
+			err := m.recreateChatSubscription(subscription.SubscriptionID, subscription.UserID, subscription.Secret)
+			if err != nil {
+				m.api.LogError("Unable to recreate chat subscription properly", "error", err)
+			}
+		} else {
+			err := m.refreshSubscription(subscription.SubscriptionID)
+			if err != nil {
+				m.api.LogError("Unable to refresh chat subscription properly", "error", err)
+			}
+		}
+	}
+}
+
+func (m *Monitor) checkGlobalSubscriptions() {
+	m.api.LogDebug("Checking for global subscriptions")
+	subscriptions, err := m.store.ListGlobalSubscriptionsToCheck()
+	if err != nil {
+		m.api.LogError("Unable to get the global subscriptions", "error", err)
+		return
+	}
+	m.api.LogDebug("Refreshing global subscriptions", "count", len(subscriptions))
+	for _, subscription := range subscriptions {
+		if time.Until(subscription.ExpiresOn) < (15 * time.Second) {
+			err := m.recreateGlobalSubscription(subscription.SubscriptionID, subscription.Type, subscription.Secret)
+			if err != nil {
+				m.api.LogError("Unable to recreate global subscription properly", "error", err)
+			}
+		} else {
+			err := m.refreshSubscription(subscription.SubscriptionID)
+			if err != nil {
+				m.api.LogError("Unable to refresh global subscription properly", "error", err)
```

Comment:
It is not strictly needed because the LogError will handle that anyway. But I think we don't have a clear policy around that. Maybe we should have one.